ACE/ace/OS_NS_sys_socket.h

```diff
@@ -276,7 +274,7 @@ namespace ACE_OS
   int shutdown (ACE_HANDLE handle,
                 int how);
 
-#if defined (ACE_LINUX) && defined (ACE_HAS_IPV6)
+#if (defined (ACE_LINUX) || defined(ACE_HAS_MAC_OSX)) && defined (ACE_HAS_IPV6)
   ACE_NAMESPACE_INLINE_FUNCTION
   unsigned int if_nametoindex (const char *ifname);
```

Comment:
In the implementation of this operation already ACE_LACKS_IF_NAME_INDEX (which I think we should rename to ACE_LACKS_IF_NAMETOINDEX) is used, so why not only compare ACE_HAS_IPV6? Looks also Windows has this operation in newer windows releases.